src/social_network_link_prediction/dimensionality_reduction_methods/svd.py

```diff
@@ -0,0 +1,65 @@
+import networkx as nx
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.sparse import csr_matrix, spdiags
+from scipy.sparse.linalg import svds
+
+def link_prediction_svd(G: nx.Graph, k = 5, normalize=False) -> csr_matrix:
+
+    # Create the adjacency matrix of the graph
+    adj_matrix = nx.adjacency_matrix(G)
```

Comment:
Va sostituita con la nostra funzione `to_adjacency_matrix`

```suggestion
    adj_matrix = to_adjacency_matrix(G)
```